### PR TITLE
Creates an objects3D variable where the components will store their c…

### PR DESCRIPTION
**Pull request by dmarcos**
…reated meshes, lights, cameras, and sounds.

### CODE REVIEW

File: src/components/camera.js [hunk 1]
```diff
@@ -16,8 +16,14 @@ module.exports.Component = registerComponent('camera', {
...
-    camera.el = el;
-    el.object3D.add(camera);
+    el.setObject3D('camera', camera);
+  },
+
+  /**
+   * Remove camera on remove (callback).
+   */
+  remove: function () {
+    this.el.setObject3D('camera');
```
Comment (ngokevin) [inline src/components/camera.js]: can we get a separate `removeObject3D`?

File: src/core/a-entity.js [hunk 2]
```diff
@@ -44,7 +44,9 @@ var proto = Object.create(ANode.prototype, {
       this.states = [];
       this.components = {};
       this.paused = true;
-      this.object3D = new THREE.Mesh();
+      this.object3D = new THREE.Group();
+      this.object3D.el = this;
+      this.objects3D = {};
```
Comment (ngokevin) [inline src/core/a-entity.js]: correct plural `object3Ds`

File: src/core/a-entity.js [hunk 3]
```diff
@@ -127,6 +129,40 @@ var proto = Object.create(ANode.prototype, {
     }
   },
 
+  getObject3D: {
+    value: function (type, Constructor) {
```
Comment (ngokevin) [inline src/core/a-entity.js]: this is the same as `getOrCreate`

File: src/core/a-entity.js (hunk 2 above)
Comment (dmarcos) [inline src/core/a-entity.js]: object3Ds is kind of weird

Comment (ngokevin) [inline src/core/a-entity.js]: `object3DMap`?

File: src/components/sound.js [hunk 4]
```diff
@@ -70,7 +70,7 @@ module.exports.Component = registerComponent('sound', {
   },
 
   remove: function () {
-    this.el.object3D.remove(this.sound);
+    this.el.setObject3D('sound', null);
```
Comment (ngokevin) [inline src/components/sound.js]: kinda feel like a `removeObject3D` would be more explicit and clear in the long run

File: src/core/a-entity.js (hunk 2 above)
Comment (ngokevin) [inline src/core/a-entity.js]: somewhat prefer `object3DMap`, `objects3D` implies an array. Or it implies "`objects`, but 3D"

**Comment by dmarcos** [comment]
Change variable name from objects3D to objects3DMap

**Comment by ngokevin** [comment]
can we do `object3DMap`?

**Pull request merged**